src/parser/view_transform.c

```diff
@@ -2162,30 +2168,29 @@ mq_update_order_by (PARSER_CONTEXT * parser, PT_NODE * statement, PT_NODE * quer
 
   statement->info.query.order_by = parser_append_node (order_by, statement->info.query.order_by);
 
-  where = statement->info.query.q.select.where;
-  if (where != NULL && PT_EXPR_INFO_IS_FLAGED (where, PT_EXPR_INFO_ROWNUM_ONLY) && statement->info.query.order_by)
+  /* generate orderby_num(), inst_num() */
+  if (!(ord_num = parser_new_node (parser, PT_EXPR)) || !(ins_num = parser_new_node (parser, PT_EXPR)))
```

Comment:
possibly memory leak if `ord_num = parser_new_node (parser, PT_EXPR)` is true and `ins_num = parser_new_node (parser, PT_EXPR)` is false.

Reply:
I fixed it.